zkevm-circuits/src/evm_circuit/execution/copy_code_to_memory.rs

```diff
@@ -0,0 +1,488 @@
+use array_init::array_init;
+use eth_types::ToLittleEndian;
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+use crate::{
+    evm_circuit::{
+        param::{N_BYTES_MEMORY_ADDRESS, N_BYTES_MEMORY_WORD_SIZE},
+        step::ExecutionState,
+        util::{
+            constraint_builder::{ConstraintBuilder, StepStateTransition, Transition},
+            math_gadget::ComparisonGadget,
+            memory_gadget::BufferReaderGadget,
+            Cell, Word,
+        },
+        witness::{Block, Call, ExecStep, StepAuxiliaryData, Transaction},
+    },
+    util::Expr,
+};
+
+use super::ExecutionGadget;
+
+/// Maximum number of bytes that can be copied in one iteration of this gadget.
+/// We are bounded because of the limitation on the number of cells we can
+/// assign to in one iteration.
+const MAX_COPY_BYTES: usize = 54;
```

Comment:
May I know how the number `MAX_COPY_BYTES` come out details ?

Reply:
So it really depends on how many cells we can allocate for each iteration. I did a binary search on this number, and for `n=55` the no. of cells in a row are insufficient. `n=54` is the most we can do in one iteration.

The idea is similar to how `CopyToMemory` gadget used `n=71`.